x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/pipelines/simulate_ml_inference_pipeline_processors.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestSimulateResponse } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import type { MlInferencePipeline } from '../../../../../common/types/pipelines';
+
+import { createApiLogic } from '../../../shared/api_logic/create_api_logic';
+
+import { HttpLogic } from '../../../shared/http';
+
+export const simulateMlInferencePipeline = async ({
+  docs,
+  indexName,
+  pipeline,
+}: {
+  docs: string;
+  indexName: string;
+  pipeline: MlInferencePipeline;
+}) => {
+  const route = `/internal/enterprise_search/indices/${indexName}/ml_inference/pipeline_processors/_simulate`;
```

Comment:
nitpick: The route should probably end with 'simulate', not with '_simulate'. The latter is an Elasticsearch convention, but we're producing RESTful(ish) APIs.